loky/backend/popen_loky_win32.py

```diff
@@ -64,14 +65,15 @@ def __init__(self, process_obj):
 
         python_exe = spawn.get_executable()
 
+        # copy the environment variables to set in the child process
+        child_env = os.environ.copy()
+        child_env.update(process_obj.env)
```

Comment:
Is the content of `process_obj.env` correct when the assertion fails under windows?

Reply:
I updated the test to print the content of `p.env` on failure and report on `xfail` so we should see if it is correct.

Reply:
Amazing report ...

```
========================== short test summary info ===========================
XFAIL tests/test_loky_backend.py::TestLokyBackend::test_child_env_process
  Can randomly fail on python2.7 and windows.
```